frappe/model/document.py

```diff
@@ -168,8 +168,13 @@ def load_from_db(self):
 			# During frappe is installed, the property "is_virtual" is not available in tabDocType, so
 			# we need to filter those cases for the access to frappe.db.get_value() as it would crash otherwise.
 			if hasattr(self, "doctype") and not hasattr(self, "module") and is_virtual_doctype(df.options):
-				self.set(df.fieldname, [])
-				continue
+				# find controller of virtual table and fetch
+				# unlike the regular API, add the parent here. The controller has no other way to figure it 
+				# out, without risking a loop.				
+				controller = get_controller(df.options)
+				data = controller.get_list({'parent_doc': self})  				
+				self.set(df.fieldname, data)
+				continue		
```

Comment:
This alone wont support virtual child tables. 


Firstly, there are following use cases:
1. Attaching a virtual child table virtual doctype.
2. Attaching a virtual child table to normal doctype. 
3. Attaching same child table to multiple doctypes. 



For all three cases, we should be able to do **at least** basic CRUD:
1. Create new rows on existing or new parent documents. 
2. Read rows when parent is read. (only this seems to be achieved)
3. Update individual row.
4. Delete individual rows.


If you can achieve this in PR then we can proceed. Else merging just partial implementation is no good. Just reading can be achieved by copying these 3-4 lines of code to relevant doctype controller too.